### PR TITLE
Centralized Infrastructure

### LINKED ISSUE #14: DNS Configuration
Register a DNS zone and automate name creation so that we can add cloudflare.  

Blocking dependency of #13 .

### PR DESCRIPTION
**Pull request by zachChilders**
This is a set of changes that will create our centralized infrastructure.  The intent here is to have some baseline deployment for a completely empty azure subscription.  This will generate new infra, keys, and storages needed by a standard deployment.  It should only be run one time by a subscription admin.

TODO:
- [x] Centralize infrastructure
- [x] Output necessary variables
- [x] Create service principal
- [x] Prevent extra runs

Closes #14 
Contributes to #41

### CODE REVIEW

File: deploy.ps1 [hunk 1]
```diff
@@ -42,14 +42,46 @@ $azureReqs = @(
...
 
+# Detect global infra
+$globalReqs = @(
+    @{
+        Name     = "Terraform init"
+        Describe = "Initialize terraform environment"
+        Test     = { Test-Path "$PSScriptRoot/tf/global/.terraform/terraform.tfstate" }
+        Set      = {
+            Set-Location -Path "tf/global"
+            terraform init
```
Comment (zachChilders) [inline deploy.ps1]: We don't have a backend yet at this point, so state is just local.

File: deploy.ps1 [hunk 2]
```diff
@@ -42,14 +42,46 @@ $azureReqs = @(
...
+    }
+)
+
 # Provision Infra
 $tfReqs = @(
     @{
         Name     = "Terraform init"
         Describe = "Initialize terraform environment"
-        Test     = { Test-Path "$PSScriptRoot/tf/.terraform/terraform.tfstate" }
+        Test     = { Test-Path "$PSScriptRoot/tf/enclave/.terraform/terraform.tfstate" }
```
Comment (zachChilders) [inline deploy.ps1:51]: An `enclave` is what I propose referring to our secured clusters as.  It implies security and sounds neat.

File: tf/.directory [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[Dolphin]
+Timestamp=2019,10,15,19,31,58
+Version=4
+
+[Settings]
+HiddenFilesShown=true
```
Comment (zachChilders) [inline tf/.directory]: This is dumb and will be removed.

File: tf/enclave/.directory [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Dolphin]
+Timestamp=2019,10,15,19,32,4
+Version=4
+
+[Settings]
+HiddenFilesShown=true
```
Comment (zachChilders) [inline tf/enclave/.directory]: This is dumb and will be removed.

File: tf/enclave/variables.tf [hunk 5]
```diff
@@ -21,7 +21,7 @@ variable resource_group_name {
 }
 
 variable location {
-  default = "Central US"
+  default = "South Central US"
```
Comment (zachChilders) [inline tf/enclave/variables.tf:24]: South Central US has an access feature I want to try out.  It's in San Antonio(ish) and fairly centralized on Azure's backbone.

File: tf/global/.directory [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[Dolphin]
+Timestamp=2019,10,15,19,32,16
+Version=4
+
+[Settings]
+HiddenFilesShown=true
```
Comment (zachChilders) [inline tf/global/.directory]: This is dumb and will be removed.

File: tf/global/global.tf [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  network_acls {
+    default_action = "Deny"
+    bypass         = "AzureServices"
+  }
+}
+
+resource "azurerm_key_vault_access_policy" "test" {
+  key_vault_id = azurerm_key_vault.global-kv.id
+
+  tenant_id = "e86183dc-d7cc-4132-8b39-a8de37272433"
```
Comment (zachChilders) [inline tf/global/global.tf]: This is our hardcoded AAD tenant.  I don't think we have a way to automatically provision one, so we'll have to provide instructions and parameterize this.

**Review by v-Zer0 — APPROVED** [review]

**Comment by zachChilders** [comment]
Also addresses #7

**Pull request merged**